### PR TITLE
numeric pole calc & compatible for more input types

### PR DESCRIPTION
**Pull request by marcelledekock**

### CODE REVIEW

**Comment by alchemyst** [comment]
I'm actually quite keen to remove sympy as a dependency since we should be able to to most of the stuff with the `tf` and `mimotf` classes. Instead of making all the functions also accept a sympy matrix, wouldn't it be better to allow conversion between them and have less dependency on sympy functions?

File: utils.py [hunk 1]
```diff
@@ -1940,6 +1955,53 @@ def minimal_realisation(a, b, c):
...
+
+    if type(A) == mimotf:
+        denom   = 1
+        num     = 1
+        
+        for j in range(A.matrix.shape[1]):
+            denom = list(numpy.poly1d(denom) * numpy.poly1d(A.matrix[0,j].denominator.coeffs))
+            num   = list(numpy.poly1d(num)   * numpy.poly1d(A.matrix[0,j].numerator.coeffs))
+            if symbolic_expr == True:
+                for n in range(len(denom)):
```
Comment (alchemyst) [inline utils.py]: This is not very idiomatic python. Have a look at [this video](https://www.youtube.com/watch?v=EnSu9hHGq5o).

File: utils.py [hunk 2]
```diff
@@ -1940,6 +1955,53 @@ def minimal_realisation(a, b, c):
...
+                    sym_den = (sym_den + denom[len(denom) - n- 1] * s**n).simplify()
```
Comment (alchemyst) [inline utils.py]: Negative indexes go from the back in Python, so `denom[len(denom)-1]` is the same as `denom[-1]`

**Comment by marcelledekock** [comment]
Could try do that...the _sym2mimotf_ function is already in utils and works quite well

**Comment by alchemyst** [comment]
I'm closing this request to give you time to work in some of those changes.

**Pull request closed without merging**